aws-rds-dbparametergroup/src/main/java/software/amazon/rds/dbparametergroup/UpdateHandler.java

```diff
@@ -41,6 +42,10 @@ protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
                 .build();
 
         return ProgressEvent.progress(request.getDesiredResourceState(), callbackContext)
+                .then(progress -> Commons.execOnceAndSaveContext(progress,
+                        () -> verifyDbParameterGroupExists(proxy, proxyClient, progress),
+                        CallbackContext::isVerifiedParameterGroupExists,
+                        CallbackContext::setVerifiedParameterGroupExists))
```

Comment:
Given this specific use-case pattern, I would suggest sticking to the framework primitives: a basic fetch call and a stabilization loop. It will provide the execOnce semantics from the box and do the fetch-and-proceed in a very unified way adopted across our resources. This will eliminate the need for the delayed progressEvent handling (`execOnceAndSaveContext`). I think this will simplify the change a bit.